Gyva/RDDS
# Implemented project model feature

## Title: Add Models and Profile Picture Handling for Supervisor and Student
 
## Description:
This PR introduces the following changes:

* Added models for Department, Faculty, Level, Supervisor, and Student.
* Implemented custom logic for generating unique reg_num patterns for Supervisor (supXXXXX) and Student (24rpXXXXX) models.
* Configured profile picture uploads for Supervisor and Student to be stored in static/supervisor/ and static/student/ directories respectively.
* Ensured that uploaded profile pictures have unique filenames using UUIDs to prevent file name conflicts.
Key Changes:

### Models:

* Created models for Department, Faculty, Level, Supervisor, and Student.
* Added necessary relationships between these models using foreign keys.
Registration Number:

* Automatically generates reg_num for Supervisor as supXXXXX.
* Automatically generates reg_no for Student as 24rpXXXXX.
* Profile Picture Uploads:

* Profile pictures are stored in static/supervisor/ and static/student/ directories.
* filenames are generated using a UUID to ensure uniqueness.

### How to Test:

* Run migrations and create new Supervisor and Student records.
* Verify that the profile pictures are stored in the correct directories with unique filenames.
* Confirm that the registration numbers follow the specified patterns.
